core/src/test/java/org/apache/calcite/test/InterpreterTest.java

```diff
@@ -579,7 +579,7 @@ private static void assertRows(Interpreter interpreter,
     final String sql = "select x, min(y), max(y), sum(y), avg(y)\n"
         + "from (values ('a', -1.2), ('a', 2.3), ('a', 15)) as t(x, y)\n"
         + "group by x";
-    sql(sql).returnsRows("[a, -1.2, 15.0, 16.1, 5.366666666666667]");
+    sql(sql).returnsRows("[a, -1.2, 15.0, 16.1, 5.3]");
```

Comment:
shouldn't in that case it be `5.4` ?
It seems rounding should be tweaked a bit
or did i miss anything?

Reply:
@snuyanzin I have written extensive comments about the rounding mode in Calcite in several places, e.g.
https://issues.apache.org/jira/browse/CALCITE-6322
I don't like this choice myself, but it's consistent in how Calcite handles rounding elsewhere.
If we want to change this we should first (1) write a spec about the semantics of casts in Calcite, (2) implement it consistently everywhere.

Reply:
I have no problem doing this work myself, but first everyone should agree on the desired result, because otherwise every piece of code in Calcite that handles conversions (and there are a lot of them) will do something different.